inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -283,6 +283,8 @@ public function save_post( $post_id ) {
 	 */
 	public function is_valid_post_type( $post_type ) {
 
+		global $wp_post_types;
```

Comment:
That would motivate me to not statically cache the results but do use the API function. If it's just a thin wrapper the PHP should manage opcode optimization (especially on higher PHP versions). I feel the arguments are kind of balancing out here, but I personally still favor using the API to use it where it's meant for.